packages/babel-parser/src/parser/statement.ts

```diff
@@ -2882,6 +2882,7 @@ export default abstract class StatementParser extends ExpressionParser {
     /* eslint-disable @typescript-eslint/no-unused-vars -- used in TypeScript and Flow parser */
     isInTypeOnlyImport: boolean,
     isMaybeTypeOnly: boolean,
+    bindingType?: BindingTypes,
```

Comment:
Can you also update `packages/babel-parser/src/plugins/flow/index.ts`? It does not need any special handling: we just have to pass through all params to the base method.

Reply:
Is there any special reason? I used optional parameters so that I can avoid modifying all plugins.

Reply:
So that we can maintain the interface consistence between the derived plugin parser and the base parser. If we started to use `bindingType` in the base parser, we are prone to forget that `flow` plugin also needs update. In the past we see several bugs related to out-of-dated `estree` plugin parser.

Reply:
I think you're right, although we can remove optional tags when we actually use them.
Seems like plugins can exist at the same time? Then I seem to need to disallow `void` here as well.
https://github.com/babel/babel/pull/14900/files#diff-37df9adada2d88868f7240c5668049abc73404f849e024e3dfd68e87a94077beR96